### PR TITLE
[kotlin-spring] Fix gradle build failure when generated with interfaceOnly=true 

### LINKED ISSUE #18803: [BUG][kotlin-spring] Gradle build fails when generated with `interfaceOnly=true`
```shell
$ gradle build -x test
> Task :bootJar FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':bootJar'.
> Error while evaluating property 'mainClass' of task ':bootJar'.
   > Failed to calculate the value of task ':bootJar' property 'mainClass'.
      > Main class name has not been configured and it could not be resolved

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
```

This happens when building locally and also on the CI pipeline: https://github.com/OpenAPITools/openapi-generator/actions/runs/9314727937/job/25639472824 – blocking any kotlin-related PRs from succeeding.

@stefankoppier @Rugal any ideas?

Strangely enough, i also tried building against the very first version locally (1e0e8fe0eb), same result.

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

[ci] [kotlin]

### PR DESCRIPTION
**Pull request by Philzen**
Resolves #18803 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by stefankoppier** [comment]
@Philzen suggested some indentation alterations. No other comments :)

File: modules/openapi-generator/src/main/resources/kotlin-spring/libraries/spring-boot/buildGradleKts.mustache [hunk 1]
```diff
@@ -20,6 +20,12 @@ tasks.withType<KotlinCompile> {
     kotlinOptions.jvmTarget = "1.8"
 }
 
+{{#interfaceOnly}}
+    tasks.bootJar {
+    enabled = false
+    }
```
[the remark this answers is not in the document]
Comment (Philzen) [reply modules/openapi-generator/src/main/resources/kotlin-spring/libraries/spring-boot/buildGradleKts.mustache]: Oh bugger, this must have slipped, thanks for spotting this!

Wondering why the CI went through as the indent does not appear in the updated samples – so i assume there's some formatting post-processing step that does this.

**Comment by Philzen** [comment]
@wing328 This is looks good to go from our side if you're up for a merge :innocent:

**Comment by Philzen** [comment]
@wing328

Now also updated the library descriptions with version numbers (all of them) as some were completely off:

https://github.com/OpenAPITools/openapi-generator/blob/d2c068cd656fe9d77c434925cb0bbde0ea886a37/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java#L241

Wonder how we could ensure this doesn't get out-of-sync the next time some dependency get's updated. Probably managing all of it in a constant, such as `JavaClientCodegen.MICROPROFILE_REST_CLIENT_DEFAULT_VERSION` already does (it is applied in the template directly).

Branch is rebased now but CI will keep failing before being rebased on #18826

**Pull request merged**